src/MyEventEmitter.js

```diff
@@ -1,12 +1,85 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (typeof listener !== 'function') {
+      throw new Error('Listener must be a function!');
+    }
+
+    this.events[eventName] = this.events[eventName] || [];
+    this.events[eventName].push(listener);
+
+    return this;
+  }
+
+  once(eventName, listener) {
+    const wrapper = () => {
+      listener();
+      this.off(eventName, wrapper);
+    };
+
+    this.events[eventName].push(wrapper);
+
+    return this;
+  }
+
+  off(eventName, listenerToRemove) {
+    if (!this.events[eventName]) {
+      throw new Error(`Error removing listener! ${eventName} does not exists!`);
+    }
+
+    const filterListeners = (listener) => listener !== listenerToRemove;
+
+    this.events[eventName] = this.events[eventName].filter(filterListeners);
+
+    return this;
+  }
+
+  emit(eventName, ...args) {
+    const fns = this.events[eventName];
+
+    if (!fns) {
+      return false;
+    }
+
+    fns.forEach(f => f(...args));
+
+    return true;
+  }
+
+  prependListener(eventName, listener) {
+    this.events[eventName] = this.events[eventName] || [];
+    this.events[eventName].unshift(listener);
+
+    return this;
+  }
+
+  prependOnceListener(eventName, listener) {
+    const wrapper = () => {
+      listener();
+      this.off(eventName, wrapper);
+    };
+
+    this.events[eventName].push(wrapper);
```

Comment:
Adds a one-time listener function for the event named eventName to the **beginning** of the listeners array.